test/e2e/api/matchMedia/map-bounding-box.test.js

```diff
@@ -0,0 +1,64 @@
+import { test, expect, chromium } from '@playwright/test';
+
+test.describe('matchMedia map-bounding-box tests', () => {
+  let page;
+  let context;
+  test.beforeAll(async () => {
+    context = await chromium.launchPersistentContext('', { headless: false });
+    page =
+      context.pages().find((page) => page.url() === 'about:blank') ||
+      (await context.newPage());
+    page = await context.newPage();
+    await page.goto('map-bounding-box.html');
+  });
+
+  test.afterAll(async function () {
+    await context.close();
+  });
+
+  test('matchMedia API detects changes in map-extents', async () => {
+    const map = page.locator('mapml-viewer');
+    const button = page.locator('.getBoundingBox');
+    const zoomIn = page.locator('.leaflet-control-zoom-in');
+    const zoomOut = page.locator('.leaflet-control-zoom-out');
+
+    await button.click();
+    const layer = page.locator('map-layer[label="test media query"]');
+
+    // upon creation, the test query layer should not be hidden
+    await expect(layer).not.toHaveAttribute('hidden');
+
+    // the test query layer should be hidden as the bound just overlaps
+    // the extent of the map
+    await zoomOut.click();
+    await page.waitForTimeout(250);
+    await zoomIn.click();
+    await expect(layer).toHaveAttribute('hidden');
+
+    // the layer should be shown as we zoom out again
+    await zoomOut.click();
+    await page.waitForTimeout(500);
+    await expect(layer).not.toHaveAttribute('hidden');
+
+    // compute starting and ending points for moving the map
+    const boundingBox = await map.boundingBox();
```

Comment:
these are fine as is, but you can also use HTMLMapmlViewerElement.zoomTo(lat,lon,zoom) to move the map and it's just as good, less trouble if you don't need the interactive stuff.